File: cap09/src/main/java/app/car/cap09/interfaces/incoming/input/TravelRequestInput.java
package app.car.cap09.interfaces.incoming.input;

import lombok.Data;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

@Data
public class TravelRequestInput {


    @NotNull(message = "O campo passengerId não pode ser nulo")
    Long passengerId;

    @NotEmpty(message = "O campo origin não pode estar em branco")
    String origin;

    @NotEmpty(message = "O campo destination não pode estar em branco")
    String destination;
}

File: cap09/src/main/kotlin/app/car/car09/interfaces/incoming/input/TravelRequestInput.kt
package app.car.car09.interfaces.incoming.input

import javax.validation.constraints.NotEmpty
import javax.validation.constraints.NotNull

data class TravelRequestInput(
        @NotNull(message = "O campo passengerId não pode ser nulo")
        val passengerId: Long,

        @NotEmpty(message = "O campo origin não pode estar em branco")
        val origin:  String,

        @NotEmpty(message = "O campo destination não pode estar em branco")
        var destination: String

)